### ISSUE TITLE
pyspark library is missing from jupyter/pyspark-notebook

### ISSUE THREAD

**Issue by Bidek56** [jupyter/docker-stacks#536]
**What docker image you are using?**
jupyter/pyspark-notebook:27ba57364579

pyspark library seems to be gone from the latest dockerhub build: jupyter/pyspark-notebook:27ba57364579

When I do:  pip list installed | grep pyspark, I get nothing.

In the prior version: jupyter/pyspark-notebook:033056e6d164
pyspark (2.2.0) exists.

What could be wrong? Thanks

**Comment by Bidek56** [comment]
When I run: docker run -it --rm -p 8889:8888 jupyter/pyspark-notebook:27ba57364579
pyspark is there.

When I run: docker run -it --rm -u root -p 8889:8888 jupyter/pyspark-notebook:27ba57364579
pyspark is gone.

Is this the expected behaviour?

**Comment by parente** [comment]
@Bidek56 When you run:

```
docker run -it --rm -p 8889:8888 jupyter/pyspark-notebook:27ba57364579 which pip
docker run -it --rm -u root -p 8889:8888 jupyter/pyspark-notebook:27ba57364579 which pip
```

what do you get as output in each case?

**Comment by Bidek56** [comment]
/opt/conda/bin/pip in both cases.

**Comment by Bidek56** [comment]
docker run -it --rm -p 8889:8888 jupyter/pyspark-notebook:27ba57364579 pip list installed | grep pyspark
pyspark (2.2.0)
docker run -it --rm -u root -p 8889:8888 jupyter/pyspark-notebook:27ba57364579 pip list installed | grep pyspark
pyspark (2.2.0)

but when I run it in the notebook, it can't find pyspark. Weird.

**Comment by Bidek56** [comment]
Any idea how to resolve this issue besides installing pyspark everytime I start the container?
Does anyone else have this issue? Thanks

**Comment by parente** [comment]
The `PYTHONPATH` set here:

https://github.com/jupyter/docker-stacks/blob/master/pyspark-notebook/Dockerfile#L37

is preserved when the container starts as the `jovyan` user:

```
docker run -it --rm -p 8888:8888 jupyter/pyspark-notebook:27ba57364579
```

When started as `root`, like so:

```
docker run -it --rm --user root -p 8888:8888 jupyter/pyspark-notebook:27ba57364579
```

the `PYTHONPATH` is lost when the `start.sh` script sudo's to `jovyan`. Interestingly, the `SPARK_HOME`, `MESOS_NATIVE_LIBRARY`, and other variables are preserved.

Looks like `/etc/sudoers` needs a bit more manipulation to restore the original behavior.

**Comment by ksingh7** [comment]
I am also hitting the exact same bug

**Comment by ksingh7** [comment]
BTW this is also hitting ``jupyter/all-spark-notebook`` image

**Comment by parente** [comment]
Happy to take a PR if someone gets to this before I do. I believe some experimentation with adding `PYTHONPATH` to the `env_keep` list in `/etc/sudoers` might do the trick. If not, we might need to simply remove the `env_reset`.

https://stackoverflow.com/questions/7969540/pythonpath-not-working-for-sudo-on-gnu-linux-works-for-root

**Comment by Bidek56** [comment]
PR #544 has been created to fix this issue.

**Comment by Bidek56** [comment]
The issue has been fixed with PR #544
